### PR TITLE
GUACAMOLE-1099: add Docker secrets support for LDAP properties

### PR DESCRIPTION
**Pull request by ss10sb**
Added _FILE for:

- LDAP_SEARCH_BIND_DN
- LDAP_SEARCH_BIND_PASSWORD
- LDAP_USER_BASE_DN
- LDAP_CONFIG_BASE_DN
- LDAP_GROUP_BASE_DN

Added SSL certificate support via LDAP_SSL_CERT_FILE (eg /run/secrets/ldap_cert) with keytool import for the file.  It could probably use an exists check but it works :)

### CODE REVIEW

**Comment by necouchman** [comment]
@ss10sb: Thank you for the contribution, appreciate it.  Per our standard procedures, all pull requests need to be associated with a JIRA issue, and tagged with that.  Please see:

http://guacamole.apache.org/open-source/

**Comment by ss10sb** [comment]
[GUACAMOLE-1099](https://issues.apache.org/jira/browse/GUACAMOLE-1099)

File: guacamole-docker/bin/start.sh [hunk 1]
```diff
@@ -363,24 +363,47 @@ END
...
+        set_optional_property "ldap-search-bind-dn" "$LDAP_SEARCH_BIND_DN"
+    fi
+
+    if [ -n "$LDAP_SEARCH_BIND_PASSWORD_FILE" ]; then
+        set_optional_property "ldap-search-bind-password" "$(cat $LDAP_SEARCH_BIND_PASSWORD_FILE)"
+    elif [ -n "$LDAP_SEARCH_BIND_PASSWORD" ]; then
+        set_optional_property "ldap-search-bind-password" "$LDAP_SEARCH_BIND_PASSWORD"
+    fi
+
+    if [ -n "$LDAP_USER_BASE_DN_FILE" ]; then
```
Comment (necouchman) [inline guacamole-docker/bin/start.sh:379]: Is there a particular reason to include the various DNs in the Docker secrets support? I understand for password, and possibly for username, but it doesn't seem like an LDAP DN is all that sensitive a value that it needs to be protected??

File: guacamole-docker/bin/start.sh [hunk 2]
```diff
@@ -394,6 +417,10 @@ END
     set_optional_property "ldap-max-referral-hops"  "$LDAP_MAX_REFERRAL_HOPS"
     set_optional_property "ldap-operation-timeout"  "$LDAP_OPERATION_TIMEOUT"
 
+    if [ -n "$LDAP_SSL_CERT_FILE" ]; then
```
Comment (necouchman) [inline guacamole-docker/bin/start.sh:420]: I'm on the fence about this - I certainly don't have any problem supporting this functionality, but I'm not sure its entirely in scope of this particular change??

**Review by necouchman — CHANGES_REQUESTED** [review]

**Review by mike-jumper — COMMENTED** [review]
Reading the environment variable `FOO` from the file `FOO_FILE` feels like the sort of thing that would be best accomplished through changes to `set_optional_property` and `set_property`, rather than repeated `if` checks.

**Comment by necouchman** [comment]
@ss10sb A handful of requests for changes, here - any chance you can polish this up so we can move forward with this? Also, there's a conflict, now, that needs to be resolved.

**Comment by ss10sb** [comment]
> @ss10sb A handful of requests for changes, here - any chance you can polish this up so we can move forward with this? Also, there's a conflict, now, that needs to be resolved.

We're no longer using Guacamole + Docker. Sorry!

**Pull request closed without merging**